lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +237,23 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        link = article_bs.find('a', class_='listing-preview__content')
+        return str(link.get('href'))
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        while len(self.urls) < self.config.get_num_articles():
+            for url in self.get_search_urls():
+                response = make_request(url, self.config)
+
+                if not response.ok:
+                    continue
+
+                soup = BeautifulSoup(response.text, 'lxml')
+                if url and url not in self.urls:
+                    self.urls.append(self._extract_url(soup))
```

Comment:
do you stop when you meet exact number of urls or you will get a bit more urls collected?

Reply:
done